Track recent files and allow opening them

vlaaad opened the pull request:
User-facing changes:

User now can press `Cmd+Shift+T` (macOS) / `Ctrl+Shift+T` (Windows and Linux) to reopen previously closed tabs. File menu now has a new submenu — Recent Files — that allows selecting a recently accessed file to open/focus on. Also, editor now has Recent Files dialog that allows searching from a larger list of recent files, available using `Ctrl+R` (macOS) / `Alt+R` (Windows and Linux) shortcuts.

Fixes #3855

vlaaad committed:
Track recent files and allow opening them

User-facing changes:

User now can press `Cmd+Shift+T` (macOS) / `Ctrl+Shift+T` (Windows and Linux) to reopen previously closed tabs. File menu now has a new submenu — Recent Files — that allows selecting a recently accessed file to open/focus on. Also, editor now has Recent Files dialog that allows searching from a larger list of recent files, available using `Ctrl+R` (macOS) / `Alt+R` (Windows and Linux) shortcuts.

Fixes #3855

vlaaad commented on editor/src/clj/editor/ui.clj:
```diff
@@ -1324,7 +1324,7 @@
                                 icon
                                 style-classes
-                                (make-menu-items scene options command-contexts {} evaluation-context)
+                                (make-menu-items scene options command-contexts command->shortcut evaluation-context)
```
Not sure why we didn't use `command->shortcut` in the submenus, but without that we didn't show shortcuts in nested submenus.

matgis reviewed (approved):
Works great! 👍

I did notice the icons in the resource picker dialog looking blurry compared to the ones in the Assets panel, but that issue seems to have been there for some time.

matgis commented:
Also, is there an easy way we can highlight the matching substrings in the filter dialog like we do in the other asset dialogs? Maybe it could be a separate PR?

<img width="276" alt="image" src="https://user-images.githubusercontent.com/11677496/190166277-f9ba57bd-1554-454b-b5e4-82aa69e528dc.png">

vlaaad committed:
Improve Recent Files dialog

This commit adds highlighting to items in the dialog similar to resource dialog's highlighting. It also makes resource icons less blurry.

vlaaad commented on editor/src/clj/editor/recent_files.clj:
```diff
@@ -0,0 +1,113 @@
+                             :spacing 4
+                             :children [{:fx/type ui/image-icon
+                                         :path (workspace/resource-icon resource)
+                                         :size 16}
```
Hmm, there was a discrepancy in behavior between `editor.ui/image-icon` and `editor.icons/get-image-view` (the latter always loaded icons in full size), I fixed it.

vlaaad commented:
@matgis I added highlighting in this PR :)
